app/services/service_provider_config.rb

```diff
@@ -0,0 +1,22 @@
+class ServiceProviderConfig
+  def initialize(filename:, issuer:)
+    @issuer = issuer
+    @data ||= YAML.load_file("#{Rails.root}/config/#{filename}")
```

Comment:
The `||=` looks pointless here?  Presumably you are trying to load the yaml file just one but does this actually do that?